lib/dynamoid/persistence.rb

```diff
@@ -152,6 +152,17 @@ def undump_field(value, options)
         end
       end
 
+      def undump_set(options, value)
+        case options[:item_type]
```

Comment:
I would call it just `type`.

`item` is used for the whole document in the DynamoDb documentation (https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.CoreComponents.html?shortFooter=true)

Reply:
Type conflicts with the main type (set, serialized, raw) etc. I didn't dig into this deeply, but I think the options hashes get merged/initialized, hence type didn't work for me. 

I went with set_collection_type. What do you think?